docs/design/coreclr/botr/exceptions.md

```diff
@@ -44,6 +44,8 @@ And here is the big difference from C++ exceptions: the CLR developer doesn't ge
 
 It bears repeating that the EX_CATCH macro catches everything. This behaviour is frequently not what a function needs. The next two sections discuss more about how to deal with exceptions that shouldn't have been caught.
 
+A function should not return from between an `EX_CATCH` usage and an `EX_END_CATCH` usage. Doing so will effectively swallow the exception.
```

Comment:
> Doing so will effectively swallow the exception.

Why is it a problem? For example, if I write the following:
``` 
bool DoStuffNoThrow()
{
    EX_TRY
    {
        DoStuff();
    }
    EX_CATCH
    {
        return false;
    }
    EX_END_CATCH(SwallowAllExceptions);

     return true;
}
```

I naturally expect the method to swallow the exception and return false if there is one. `SwallowAllExceptions` is redundant in this case, but it should not affect correctness.

Or are there other negative side-effects if I do that?

Reply:
My main concern here was I wanted docs to point to in case someone uses a something like `RethrowTransientExceptions` but has an early return.

Reply:
Should we get rid of the argument for `EX_END_CATCH` macro and make rethrowing a statement to make it more obvious that it is part of the regular flow and returning before that won't execute it?

```
    EX_CATCH
    {
        return false;
        RethrowTransientExceptions;
    }
    EX_END_CATCH;
```

Some existing places do it like that already, For example: https://github.com/dotnet/runtime/blob/15c514180982522386424c7bface57564ad6724d/src/coreclr/vm/clsload.cpp#L1127